common/lib/xmodule/xmodule/modulestore/courseware_index.py

```diff
@@ -0,0 +1,117 @@
+""" Code to allow module store to interface with courseware index """
+from __future__ import absolute_import
+
+import logging
+
+from opaque_keys.edx.locator import CourseLocator
+from search.search_engine_base import SearchEngine
+
+from . import ModuleStoreEnum
+from .exceptions import ItemNotFoundError
+
+# Use default index and document names for now
+INDEX_NAME = "courseware_index"
+DOCUMENT_TYPE = "courseware_content"
+
+log = logging.getLogger('edx.modulestore')
+
+
+class ModuleStoreCoursewareIndexMixin(object):
+    """
+    Mixin class to enable indexing for courseware search from different modulestores
+    """
+
+    def do_index(self, location, delete=False):
+        """
+        Add to courseware search index from given location and its children
+        """
+        error = []
+        # TODO - inline for now, need to move this out to a celery task
+        searcher = SearchEngine.get_search_engine(INDEX_NAME)
+        if not searcher:
+            return
+
+        course_key = location if isinstance(location, CourseLocator) else location.course_key
```

Comment:
Is there a reason you have this all on one line? I feel like it would be more readable to break it up:
```python
if isinstance(location, CourseLocator):
    course_key = location
else:
    course_key = location.course_key
```